docs/docs/programmatically-create-pages-from-data.md

```diff
@@ -21,35 +21,29 @@ access to the
 [`createPage`](/docs/actions/#createPage) action
 which is at the core of programmatically creating a page.
 
-```javascript:title=gatsby-node.js
-exports.createPages = ({ graphql, actions }) => {
-  const { createPage } = actions
-  return new Promise((resolve, reject) => {
-    graphql(`
-      {
-        allMarkdownRemark {
-          edges {
-            node {
-              fields {
-                slug
-              }
+```js:title=gatsby-node.js
+exports.createPages = async function({ actions, graphql }) {
+  // highlight-next-line
+  await graphql(`
+    {
+      allMarkdownRemark {
+        edges {
+          node {
+            fields {
+              slug
             }
           }
         }
       }
-    `).then(result => {
-      // highlight-start
-      result.data.allMarkdownRemark.edges.forEach(({ node }) => {
-        createPage({
-          path: node.fields.slug,
-          component: path.resolve(`./src/templates/blog-post.js`),
-          context: {
-            slug: node.fields.slug,
-          },
-        })
+    }
+  `).then(res => {
+    res.data.allMarkdownRemark.edges.forEach(edge => {
```

Comment:
Perhaps move this after the await? It tends to clean up the code if we can move off of .then blocks!

e.g.

```
const { data } = await graphql(`
 # query
`)

data.allMarkdownRemark.edges.forEach(edge => {
```

Reply:
@DSchau I made the changes and tested it, works for me. Added the highlighting as it was in the previous code sample.